reducer/src/main/java/com/graphicsfuzz/reducer/ReductionDriver.java

```diff
@@ -424,7 +424,7 @@ private void writeState(ShaderJob state, File shaderJobFileOutput,
     ShaderJob stateToWrite = state.clone();
     if (requiresUniformBindings) {
       assert !stateToWrite.hasUniformBindings();
-      stateToWrite.makeUniformBindings();
+      stateToWrite.makeUniformBindings(Optional.empty());
```

Comment:
This will require a bit of care.

If requiresUniformBindings holds, the reducer removes uniform bindings before transforming a shader and adds them back before writing it back out.

We will need to add an Optional<String>, pushConstant, to track the push constant that was bound (if any), alongside requiresUniformBindings, and then pass it here.

Reply:
Done and took quite a few changes.